shiny/render/_data_frame_utils/_selection.py

```diff
@@ -0,0 +1,450 @@
+    x: CellSelection | BrowserCellSelection,
+) -> BrowserCellSelection:
+
+    if x == "all":
+        return {"type": "all"}
+    if x == "none":
+        return {"type": "none"}
+
+    # Can't handle lists of dictionaries right now
+    assert isinstance(x, dict)
+
+    if "type" in x:
+        # Looks like a BrowserCellSelection, verify and return
+        if x["type"] == "all":
+            return x
+        if x["type"] == "none":
+            return x
+        if x["type"] == "row":
+            assert "rows" in x
+            assert isinstance(x["rows"], Sequence)
+            return x
+        if x["type"] == "col":
+            assert "cols" in x
+            assert isinstance(x["cols"], Sequence)
+            return x
+        if x["type"] == "region":
+            assert "rows" in x
+            assert "cols" in x
+            assert isinstance(x["rows"], Sequence)
+            assert isinstance(x["cols"], Sequence)
+            return x
+        raise ValueError(f"Unhandled BrowserCellSelection['type']: {x['type']}")
+
+    # Not a BrowserCellSelection, convert as if a CellSelection!
+    row_value = to_list_or_none(x.get("rows", None))
+    col_value = to_list_or_none(x.get("cols", None))
+
+    if row_value is None:
+        if col_value is None:
+            # None!
+            return {"type": "none"}
+        else:
+            # Col!
+            assert isinstance(col_value, Sequence)
+            if len(col_value) == 0:
+                # Nothing selected
+                return {"type": "none"}
+            return {"type": "col", "cols": tuple(col_value)}
+    else:
+        # row_value is not None
+        if col_value is None:
+            # Row!
+            assert isinstance(row_value, Sequence)
+            if len(row_value) == 0:
+                # Nothing selected
+                return {"type": "none"}
+
+            return {"type": "row", "rows": tuple(row_value)}
+        else:
+            # Region!
+            assert isinstance(row_value, Sequence)
+            assert isinstance(col_value, Sequence)
+            assert len(row_value) == 2
+            assert len(col_value) == 2
+            return {
+                "type": "region",
+                "rows": (row_value[0], row_value[1]),
+                "cols": (col_value[0], col_value[1]),
+            }
+
+
+RowSelectionModeDeprecated = Literal["single", "multiple", "none", "deprecated"]
+
+
+def as_selection_mode(
+    selection_mode: SelectionMode,
+    *,
+    name: str,
+    editable: bool,
+    row_selection_mode: RowSelectionModeDeprecated = "deprecated",
+) -> SelectionMode:
+    if row_selection_mode == "deprecated":
+        # Disable selection_mode if `editable=True``
+        if editable:
+            warnings.warn(
+                '`editable` can not be `True` while `selection_mode` != `"none"`. '
+                'Setting `selection_mode="none"`',
+                stacklevel=2,
+            )
+            return "none"
```

Comment:
I'm not sure I fully understand the logic of how `row_selection_mode` is used here... Can you clarify with a comment?

But regarding the warning: this is temporary, right? At some point we will want to allow row selection and editing.